sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -193,6 +184,21 @@ export class ApplicationStudentsController extends BaseController {
       // then study start date taken from offering
       studyStartDate = offering.studyStartDate;
       studyEndDate = offering.studyEndDate;
+      // This ensures that if an offering is selected in student application,
+      // a wrong study start date and study end date which doesn't match
+      // the selected offering id is sent to API by modifying the payload.
+      payload.data.selectedOfferingDate = studyStartDate;
```

Comment:
I noticed that the logic seems to be duplicated in the form.io on hidden fields in the same form:
- Study end date past maximum allowed days (studyEndDatePastAllowedDays)
- Selected study end date past maximum allowed days (selectedStudyEndDatePastAllowedDays)
I have two observations about it:
1. Since the logic is also based on the `data.studyendDate` I believe that this value also needs to be resigned in the server, right?
2. Is it not possible to have just one hidden field with the logic?

Reply:
Thank you for the clarification over the call.